ouroboros-network/test/Test/Mux.hs

```diff
@@ -390,61 +517,275 @@ prop_mux_starvation response0 response1 =
+        -- single segment.
+        let server_small ChainSyncSmall = server_mp
+
+        (client_w, said) <- plainServer server_small
+        setup state client_w
+
+        writeSdu client_w $! unDummyPayload sdu
+
+        atomically $! putTMVar stopVar $ unDummyPayload sdu
 
-    (_, _, server_mp) <- setupMiniReqRsp (return ()) endMpsVar request response
-    let server_mps ChainSync1 = server_mp
+        res <- wait said
+        case res of
+            Just e  ->
+                case fromException e of
+                    Just me -> return $ Mx.errorType me === Mx.MuxIngressQueueOverRun
+                    Nothing -> return $ property False
+            Nothing -> return $ property False
+
+    run (ArbitraryValidSDU sdu state err_m) = do
+        stopVar <- newEmptyTMVarM
+
+        let server_mp = Mx.MiniProtocolDescription Nothing (Just $ serverRsp stopVar)
+        let server_std ChainSync1 = server_mp
+
+        (client_w, said) <- plainServer server_std
+
+        setup state client_w
+
+        atomically $! putTMVar stopVar $! unDummyPayload sdu
+        writeSdu client_w $ unDummyPayload sdu
+
+        res <- wait said
+        case res of
+            Just e  ->
+                case fromException e of
+                    Just me -> case err_m of
+                                    Just err -> return $ Mx.errorType me === err
+                                    Nothing  -> return $ property False
+                    Nothing -> return $ property False
+            Nothing -> return $ err_m === Nothing
+
+    run (ArbitraryInvalidSDU badSdu state err) = do
+        stopVar <- newEmptyTMVarM
+
+        let server_mp = Mx.MiniProtocolDescription Nothing (Just $ serverRsp stopVar)
+        let server_std ChainSync1 = server_mp
+
+        (client_w, said) <- plainServer server_std
+
+        setup state client_w
+        atomically $ writeTBQueue client_w $ BL.take (isRealLength badSdu) $ encodeInvalidMuxSDU badSdu
+        atomically $ putTMVar stopVar BL.empty
+
+        res <- wait said
+        case res of
+            Just e  ->
+                case fromException e of
+                    Just me -> return $ Mx.errorType me === err
+                    Nothing -> return $ property False
+            Nothing -> return $ property False
+
+    plainServer server_mps = do
+        server_w <- atomically $ newTBQueue 10
+        server_r <- atomically $ newTBQueue 10
+
+        said <- startMuxSTM server_mps server_w server_r 1280 Nothing
+
+        return (server_r, said)
+
+    -- Server that expects to receive a specific ByteString.
+    -- Doesn't send a reply.
+    serverRsp stopVar chan =
+        atomically (takeTMVar stopVar) >>= loop
+      where
+        loop e | e == BL.empty = return ()
+        loop e = do
+            msg_m <- recv chan
+            case msg_m of
+                 Just msg ->
+                     let e_m = BL.stripPrefix msg e in
+                     case e_m of
```

Comment:
why not just `case BL.stripPrefix msg e of`

Reply:
Will address.